### PR TITLE
Add tasks, refactor hosts+interfaces, improve error handling

### LINKED ISSUE #62: Task scheduling
Pycroft should be able to execute asynchronous tasks at a given time.

This is not necessary for suspending a membership thanks to the intervalled membership logic, but for things like changing the mac address or the room a user lives in, schedule execution at a given time would drastically improve things for our members.

### PR DESCRIPTION
**Pull request by ibot3**
This PR adds and improves some things:

**Scheduled Tasks** 7445756
Tasks like moving, moving in, moving out a user can be scheduled to a specific date.
This is done with Celery. I introduced a new docker service, the "celery-worker".
Closes #62 

**(UI) Merging the interfaces table into the hosts table** c116ab7
The table to show the interfaces of a host is displayed as a subtable when clicking on a host row.
Further, all depending web functions for hosts and interfaces were moved to a new package called "host".
**This caused many "new" lines which aren't new but were just moved. Mostly from user/\_\_init\_\_.py to host/\_\_init\_\_.py**

**Error handling improvements** c71c47d
To avoid 5xx errors for users, all exceptions will be catched and an corresponding error message will be displayed. This isn't yet implemented everywhere.
References #288 and #240

### CODE REVIEW

File: pycroft/lib/task.py [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        task_impl = task_type_to_impl.get(task.type)()
+
+        try:
+            task_impl.execute(task)
+        except Exception as e:
+            task_impl.errors.append(str(e))
+
+        repair_session()
+
+        if task_impl.new_status:
```
Comment (Georgeto) [inline pycroft/lib/task.py]: Shouldn't this be `if task_impl.new_status is not None:`?

File: web/blueprints/helpers/exception.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        # Special case: Username already taken in Abe
+        # Should be removed after migration
+        if e.message == "Username already taken in Abe!":
+            flash("Dieser Benutzername wird bereits in Abe verwendet.", 'error')
+
+            session.session.rollback()
+        else:
+            raise e
+    except Exception as e:
+        traceback.print_exc()
```
Comment (ibot3) [inline web/blueprints/helpers/exception.py]: I am note sure if this actually works, should be tested.

File: pycroft/model/task.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    parameters_json = Column('parameters', JSONB, nullable=False)
+    created = Column(DateTimeTz, nullable=False)
+    creator = relationship('User')
+    creator_id = Column(Integer, ForeignKey("user.id"), nullable=False)
+    status = Column(Enum(TaskStatus), nullable=False, default=TaskStatus.OPEN)
+    errors = Column(JSONB, nullable=True)
+
+    @hybrid_property
+    def parameters(self):
+        from pycroft.lib.task import task_type_to_impl
```
Comment (Georgeto) [inline pycroft/model/task.py]: The model should not import stuff from the library.

File: pycroft/model/task.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            raise ValueError("cannot find schema for task type")
+
+        parameters_schema = task_type_to_impl[self.type].schema()
+
+        return parameters_schema.load(self.parameters_json)
+
+
+    @parameters.setter
+    def parameters(self, _parameters):
+        from pycroft.lib.task import task_type_to_impl
```
Comment (Georgeto) [inline pycroft/model/task.py]: The model should not import stuff from the library.

**Pull request merged**